apps/arena/lib/arena/game_updater.ex

```diff
@@ -1552,24 +1552,15 @@ defmodule Arena.GameUpdater do
       name: "Circle 1"
     }
 
-    entities_to_collide_with =
+    static_obstacles =
       Map.filter(obstacles, fn {_obstacle_id, obstacle} -> obstacle.aditional_info.type == "static" end)
-      |> Map.merge(%{external_wall.id => external_wall})
-
-    external_wall_id = external_wall.id
-
-    case Physics.check_collisions(circle, entities_to_collide_with) do
-      [^external_wall_id | []] ->
```

Comment:
Why was this check here? Are we losing an edge case handler by removing this?

Reply:
This was something used by the recursive implementation and caused trouble in both errors:
- Infinite loop
- Items spawning outside of the map